plugin/al/lib/AL_USDMaya/CMakeLists.txt

```diff
@@ -182,6 +182,15 @@ target_include_directories(
     $<$<BOOL:${UFE_FOUND}>:${UFE_INCLUDE_DIR}>
 )
 
+# There were bugs in both cmake and findboost with the boost version string. So in
+# order to support all boost and cmake versions, we check how the boost version is
+# defined and create our version comparison string.
+if(${Boost_VERSION} MATCHES "^([0-9]+)\\.([0-9]+)\\.([0-9]+)$")
+    set(boost_1_70_0_ver_string "1.70.0")
+else()
+    set(boost_1_70_0_ver_string "107000")
+endif()
```

Comment:
USD uses a different version of boost on Windows (1.70.0) and Linux/Mac, python 2 (1.66.0) and python 3 (1.70.0). Because of this the Boost_VERSION string has a different format between the two versions. And Boost_VERSION_STRING is not always set. There is also a fix in cmake 3.15, but we support 3.13-3.17.

    -- Boost_VERSION = 106100
    -- Boost_VERSION_STRING = 1.61.0

    -- Boost_VERSION = 1.70.0
    -- Boost_VERSION_STRING =

So in order to work around all of this I detect the format of the boost version and set a string for 1.70.0 to compare against.